### PR TITLE
Fix share to dc

### LINKED ISSUE #1249: share-to-delta from system-gallery stopped working
probably a regression from the recent changes wrt stickers :)

### PR DESCRIPTION
**Pull request by cyBerta**
I basically stubled into the same issue as here: https://github.com/deltachat/deltachat-ios/pull/1246#issue-659165534

closes #1249

### CODE REVIEW

File: deltachat-ios/Helper/ImageFormat.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        case 0x42 where data.count >= 2 :
+            if data[1] == 0x4D {
+                return .bmp
+            }
+
+        default:
+            break
+        }
+        return .unknown
```
Comment (r10s) [inline deltachat-ios/Helper/ImageFormat.swift:45]: where does this list come from?
maybe add a link to the source (if any).

File: deltachat-ios/Helper/ImageFormat.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    static func loadImageFrom(data: Data) -> UIImage? {
+        switch ImageFormat.get(from: data) {
+        case .gif, .png, .webp, .heic:
```
Comment (r10s) [inline deltachat-ios/Helper/ImageFormat.swift:50]: `.heic` can be animated as well? because of the "live-image"?

**Review by r10s — APPROVED** [review]
lgtm.

it happend once to me that that sharing did not work at the first place: the "one moment..." dialog was displayed but vanished too fast without an error - and no image was finally shared. but i could not reproduce it - so that might be another bug and should not block this pr from merging.

**Comment by r10s** [comment]
when looking at `SDAnimatedImage` source, it says:

"The image class which supports animating on `SDAnimatedImageView`. You can also use it on normal UIImageView/NSImageView."

so, maybe we can just always use `SDAnimatedImageView` unconditionally - or is there a reasonable speed advantage in using `UImage` for jpg?

File: deltachat-ios/Helper/ImageFormat.swift (hunk 2 above)
Comment (cyBerta) [reply deltachat-ios/Helper/ImageFormat.swift:50]: yes: https://nokiatech.github.io/heif/comparison.html

File: deltachat-ios/Helper/ImageFormat.swift (hunk 1 above)
Comment (cyBerta) [reply deltachat-ios/Helper/ImageFormat.swift:45]: https://en.wikipedia.org/wiki/List_of_file_signatures

Comment (cyBerta) [reply deltachat-ios/Helper/ImageFormat.swift:45]: I'll can add that as a comment as well

**Comment by cyBerta** [comment]
> when looking at `SDAnimatedImage` source, it says:
> 
> "The image class which supports animating on `SDAnimatedImageView`. You can also use it on normal UIImageView/NSImageView."
> 
> so, maybe we can just always use `SDAnimatedImageView` unconditionally - or is there a reasonable speed advantage in using `UImage` for jpg?

Yes, I tried to use SDAnimatedImage on UIImage, but unfortunaltely if you try to parse Data from a jpeg and try to show it in SDAnimatedImageView or UIImageView, it didn't work. It let to the regression that has been fixed with this PR. 
I bumped into the same problem when working on async loading of images in GalleryItem so that I had to distinguish between different formats.

**Comment by cyBerta** [comment]
> "The image class which supports animating on `SDAnimatedImageView`. You can also use it on normal UIImageView/NSImageView."

Maybe this is a bug in SDAnimatedImage

File: deltachat-ios/Helper/ImageFormat.swift (hunk 1 above)
Comment (r10s) [reply deltachat-ios/Helper/ImageFormat.swift:45]: yip, i would add that link as a comment.

**Comment by r10s** [comment]
k, then the distinction between `SDAnimatedImage` and `UIImage` makes sense, at least for now. however, maybe also add your findings to a comment atop of `loadImageFrom()`, saying, that it theoretically should work but it practically does not.

File: deltachat-ios/Helper/ImageFormat.swift (hunk 1 above)
Comment (cyBerta) [reply deltachat-ios/Helper/ImageFormat.swift:47]: done

**Review by r10s — APPROVED** [review]
great!

**Pull request merged**